broadinstitute/pearl
# JN-336 data dictionary export

This brings over the data dictionary export functionality from Pepper.  As with export, the code is much less important than thinking about the underlying data concepts.  A data dictionary overlaps *a lot* with survey definitions, but is also distinct.  Refining this dictionary export over the next few months is (IMHO) the surest bet to making sure that our survey question models are robust and well-documented.  Lots of thorny questions like "is a question numeric if all the possible answers are numeric, except for one 'prefer not to answer'"?  

Unfortunately, this only supports Excel export for now, since the goal for Pepper was to get something into study staff's hands as quickly as possible, and Excel worked way better for them than other formats.  We'll want to add tsv export soon, if for no other reason than to enable more robust testing (e.g. making sure the row headers of the dictionary match the column headers of the enrollee export).  Currently, the only test is to make sure the export works and writes something.

TO TEST:
1. restart apiAdminApp
2. go to `https://localhost:3000/ourhealth/studies/ourheart/env/sandbox/participants`
3. click 'export dictionary'
4. confirm you get an excel file that lists the profile fields and survey questions that would be exported.
